src/pyhf/workspace.py

```diff
@@ -267,7 +267,12 @@ def __init__(self, spec, **config_kwargs):
         """Workspaces hold the model, data and measurements."""
         super(Workspace, self).__init__(spec, channels=spec['channels'])
         self.schema = config_kwargs.pop('schema', 'workspace.json')
-        self.version = config_kwargs.pop('version', None)
+
+        if 'version' in spec:
+            self.version = spec['version']
+        else:
+            self.version = config_kwargs.pop('version', None)
+
```

Comment:
this should be done after schema validation which guarantees a `version` in the spec.